### PR TITLE
Destiny symbols support in names, notes and hashtags

### PR DESCRIPTION
**Pull request by robojumper**
Depends on DestinyItemManager/d2-additional-info#428.

grafik

grafik

grafik

### CODE REVIEW

File: src/app/item-popup/NotesArea.tsx [hunk 1]
```diff
@@ -115,28 +116,30 @@ function NotesEditor({
...
-        maxLength={maxLength}
-        value={liveNotes}
-        onClick={onClick}
-        onChange={onNotesUpdated}
-        onBlur={stopEvents}
-        onKeyDown={onKeyDown}
-        onTouchStart={stopEvents}
-        onMouseDown={stopEvents}
-      />
+      <div style={{ display: 'flex' }}>
```
Comment (bhollis) [inline src/app/item-popup/NotesArea.tsx:126]: I generally prefer a class

File: src/app/loadout-drawer/LoadoutDrawer2.tsx [hunk 2]
```diff
@@ -234,13 +235,16 @@ export default function LoadoutDrawer2({
...
       <details className={styles.notes} open={Boolean(loadout.notes?.length)}>
         <summary>{t('MovePopup.Notes')}</summary>
-        <TextareaAutosize
-          onChange={handleNotesChanged}
-          ref={ref}
-          value={loadout.notes}
-          maxLength={2048}
-          placeholder={t('Loadouts.NotesPlaceholder')}
-        />
+        <div style={{ display: 'flex' }}>
```
Comment (bhollis) [inline src/app/loadout-drawer/LoadoutDrawer2.tsx:238]: Maybe we need a component that wraps a textarea plus this button?

File: src/app/dim-ui/destiny-symbols/SymbolsPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+              />
+            </ClickOutside>
+          </div>,
+          pressTipRoot.current || tempContainer
+        )}
+    </div>
+  );
+}
+
+function SymbolsWindow({ onChooseGlyph }: { onChooseGlyph: (unicode: string) => void }) {
```
Comment (bhollis) [inline src/app/dim-ui/destiny-symbols/SymbolsPicker.tsx:83]: I suspect you could `memo` this (if you used `useCallback` on `onChooseGlyph`).

File: src/app/loadout-drawer/LoadoutDrawer2.tsx (hunk 2 above)
Comment (robojumper) [reply src/app/loadout-drawer/LoadoutDrawer2.tsx]: Yeah, should be better this way.

**Review by bhollis — APPROVED** [review]
Super exciting!

**Pull request merged**